cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -40,12 +40,13 @@ if(ARROW_CPU_FLAG STREQUAL "x86")
     set(CXX_SUPPORTS_SSE4_2 TRUE)
   else()
     set(ARROW_SSE4_2_FLAG "-msse4.2")
-    set(ARROW_AVX2_FLAG "-mavx2")
+    set(ARROW_AVX2_FLAG "-march=core-avx2")
```

Comment:
hopefully this is acceptable change.  I'm open to other suggestions on how to enable bmi2 (not for some reason using "-mbmi2 -mavx2" didn't seem to work for me, but I can try again.

Reply:
It would be nice if you could try again.

Reply:
done, seems to work.